include/net/net_if.h

```diff
@@ -821,25 +821,56 @@ int net_if_config_ipv6_put(struct net_if *iface);
 /**
  * @brief Check if this IPv6 address belongs to one of the interfaces.
  *
- * @param addr IPv6 address
+ * @param addr Pointer to IPv6 address
  * @param iface Pointer to interface is returned
  *
  * @return Pointer to interface address, NULL if not found.
  */
 struct net_if_addr *net_if_ipv6_addr_lookup(const struct in6_addr *addr,
 					    struct net_if **iface);
 
+/**
+ * @brief Check if this IPv6 address belongs to one of the interfaces.
+ *
+ * @param addr IPv6 address
+ * @param iface Pointer to interface is returned
+ *
+ * @return Pointer to interface address, NULL if not found.
+ */
+static inline
+struct net_if_addr *net_if_ipv6_addr_lookup_by_value(
```

Comment:
This is weird signature. Why not return `struct net_if *` directly, out of which address can be gotten as needed, and get rid of "out" params?